### PR TITLE
[FLINK-6431] [metrics] Activate strict checkstyle in flink-metrics

### PR DESCRIPTION
**Pull request by zentol**

### CODE REVIEW

File: flink-metrics/flink-metrics-datadog/src/main/java/org/apache/flink/metrics/datadog/DatadogHttpReporter.java [hunk 1]
```diff
@@ -146,44 +146,38 @@ public void report() {
...
 
-	/**
-	 * Get host from MetricGroup#getAllVariables() if it exists; returns Null otherwise
-	 * */
 	private String getHostFromMetricGroup(MetricGroup metricGroup) {
 		return metricGroup.getAllVariables().get(HOST_VARIABLE);
 	}
 
-	/**
-	 * Given "<xxx>", return "xxx"
```
Comment (greghogan) [inline flink-metrics/flink-metrics-datadog/src/main/java/org/apache/flink/metrics/datadog/DatadogHttpReporter.java]: Could be helpful to leave a comment noting the removal of the leading and trailing angle brackets.

File: flink-metrics/pom.xml [hunk 2]
```diff
@@ -60,4 +60,42 @@ under the License.
...
+					<dependency>
+						<groupId>com.puppycrawl.tools</groupId>
+						<artifactId>checkstyle</artifactId>
+						<version>6.19</version>
+					</dependency>
+				</dependencies>
+				<configuration>
+					<configLocation>../tools/maven/strict-checkstyle.xml</configLocation>
+					<suppressionsLocation>../tools/maven/suppressions.xml</suppressionsLocation>
+					<logViolationsToConsole>true</logViolationsToConsole>
```
Comment (greghogan) [inline flink-metrics/pom.xml]: includeTestSourceDirectory?

File: flink-metrics/pom.xml [hunk 3]
```diff
@@ -60,4 +60,42 @@ under the License.
...
+				<version>2.17</version>
+				<dependencies>
+					<dependency>
+						<groupId>com.puppycrawl.tools</groupId>
+						<artifactId>checkstyle</artifactId>
+						<version>6.19</version>
+					</dependency>
+				</dependencies>
+				<configuration>
+					<configLocation>../tools/maven/strict-checkstyle.xml</configLocation>
```
Comment (greghogan) [inline flink-metrics/pom.xml]: Use "absolute" paths as `flink-streaming-java`?

Comment (zentol) [inline flink-metrics/pom.xml]: huh, I'm surprised that the configuration in streaming-java actually works.

**Comment by zentol** [comment]
I've included the sources, change the path and added the removed comment. Moving on to the other 2 checkstyle PRs...

**Comment by greghogan** [comment]
+1 to merge after Travis gives the green light.

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**